core/include/scipp/core/transform_common.h

```diff
@@ -41,9 +41,10 @@ using arithmetic_type_pairs_with_bool =
     decltype(std::tuple_cat(std::declval<arithmetic_type_pairs>(),
                             std::declval<pair_numerical_with_t<bool>>()));
 
-using arithmetic_and_matrix_type_pairs = decltype(
-    std::tuple_cat(std::declval<arithmetic_type_pairs>(),
-                   std::tuple<std::tuple<Eigen::Vector3d, Eigen::Vector3d>>()));
+using arithmetic_and_matrix_type_pairs = decltype(std::tuple_cat(
+    std::declval<arithmetic_type_pairs>(),
+    std::tuple<scipp::core::time_point, scipp::core::time_point>(),
```

Comment:
This should not be here, and explains why you had to make the changes in `+` which should not be there. Time-points are *not* arithmetic types.

Reply:
fixed